VK_EXT_extended_dynamic_state validation

pdaniell-nv opened the pull request:
Task list is here: https://github.com/KhronosGroup/Vulkan-Docs/issues/1297

pdaniell-nv commented:
@mark-lunarg Would you mind taking a look at this? Thanks.

mark-lunarg commented on layers/core_validation.cpp:
```diff
@@ -724,6 +736,47 @@ bool CoreChecks::ValidatePipelineDrawtimeState(const LAST_BOUND_STATE &state, co
             }
         }
+
+        bool dynViewportCount = IsDynamic(pPipeline, VK_DYNAMIC_STATE_VIEWPORT_WITH_COUNT_EXT);
```
Local vars need to be lower_snake_case, here and several places below...

pdaniell-nv replied:
It has been hard for me to determine what is the correct style. In this case I simply copied the style of the local variables above like "bool dynViewport" and "const auto requiredScissorMask". Should I still change it to snake style even though it would be inconsistent with the above?

mark-lunarg replied:
I'll leave it up to you.  We haven't had time to scrub the codebase so it is very confusing, and we've been relying on code reviews to keep things headed in the right direction.  Maybe someday we'll get the time...

pdaniell-nv commented:
> Looks great so far, small nit about a few local var names. When this is baked, all these changes can be squashed, and we'll need a commit with the tests for each class of the validation checks (not necessarily for each Cmd, but at least one for each of the VUID types). Thanks!

Tests are now implemented. They found a few VU minor implementation bugs.

mark-lunarg commented on tests/vklayertests_others.cpp:
```diff
@@ -27,6 +27,8 @@
 #include "cast_utils.h"
 #include "layer_validation_tests.h"
 
+#define ARRAY_SIZE(a) (sizeof(a) / sizeof(a[0]))
```
I this used anyplace?

mark-lunarg commented:
@pdaniell-nv, running this through CI now.  I will need the tests split into their own commit, but I can take care of that.  We'd also prefer to avoid using C pre-processor macros such as `GET_EDS_FUNCTION`.  The commented out codeblocks that say 'Needs multiview disabled' -- is this because other errors are generated, or that it's enabled on most hardware?  Maybe we can work around the issue.  Thanks! And I will make sure this makes the upcoming SDK release.

pdaniell-nv commented:
> @pdaniell-nv, running this through CI now. I will need the tests split into their own commit, but I can take care of that.

Thanks.

> We'd also prefer to avoid using C pre-processor macros such as `GET_EDS_FUNCTION`.

Fixed.

> The commented out codeblocks that say 'Needs multiview disabled' -- is this because other errors are generated, or that it's enabled on most hardware?

I added a new variant to cover those with multiViewport feature disabled.

> And I will make sure this makes the upcoming SDK release.

Great, thanks.

mark-lunarg commented:
Made some format/style changes in the commit messages, ran through CI, and pushed to Master.  Thanks @pdaniell-nv.

pdaniell-nv commented:
> Made some format/style changes in the commit messages, ran through CI, and pushed to Master. Thanks @pdaniell-nv.

Thanks!
